rust-doc-style-guide

```diff
@@ -0,0 +1,268 @@
+
+  > Using C or C++ inside of a Rust project consists of two major parts:
+
+  > - Wrapping the exposed C API for use with Rust
+  > - Building your C or C++ code to be integrated with the Rust code
+
+- If a bullet point is a full sentence, you can end it with a full stop.
+  - If a list has at least one full stop, end all other list items with a full stop.
+
+  > To reliably interact with these peripherals:
+
+  > - Always use `volatile` methods to read or write to peripheral memory, as it can change at any time.
+  > - In software, we should be able to share any number of read-only accesses to these peripherals.
+  > - If some software should have read-write access to a peripheral, it should hold the only reference to that peripheral.
+
+- For longer list items, consider using a summary word of phrase to make content [scannable](https://learn.microsoft.com/en-us/style-guide/scannable-content/).
+
+  > If you run Windows on your host machine, make sure ...
+  >
+  > - **MSVC**: Recommended ABI, included in ...
+  > - **GNU**: ABI used by the GCC toolchain ...
+
+  -  For an example using bold font, see the list in the [Modules Cheat Sheet](https://doc.rust-lang.org/book/ch07-02-defining-modules-to-control-scope-and-privacy.html#modules-cheat-sheet) section in The Rust Programming Language book.
+  -  For an example using monospace font, see the [Panicking](https://docs.rust-embedded.org/book/start/panicking.html#panicking) section in The Embedded Rust Book.
+
+
+## Using `monospace`
+
+Use monospace font for the following items:
+
+- Code snippets
+
+  The [Installation](https://doc.rust-lang.org/book/ch01-01-installation.html) chapter in the Rust Programming Language book suggests:
+
+  - Start the terminal commands with `$`
+  - Output of previous commands should not start with `$`
+  - For PowerShell-specific examples, use `>` instead of `$`
+
+  [book-ja](https://github.com/rust-lang-ja/book-ja/blob/master-ja/style-guide.md) also suggests:
+
+  - Use `bash` syntax highlighting
+
+- Rust declarations: commands, functions, arguments, parameters, flags, variables
+- In-line command line output
+
+  > Writing a program that prints `Hello, world!`
+
+- Data types: `i8`, `u128`
+- Names of crates, traits, libraries,
+- Command line tools, plugins, packages
+- Files: `config.toml`
+- Architectures and targets
+
+ > The `RISC-V` and `Xtensa` architectures.
+
+### Monospace and Other Types of Formatting
+
+Monospace font can also be used in:
+
+- links
+
+  > [`String`](https://doc.rust-lang.org/std/string/struct.String.html) is a string type provided by ...
+
+- headings
+
+  > ### A `no_std` Rust Environment
+
+
+## Using _Italics_
+
+- Introduce new terms
+
+  > Each value in Rust has an _owner_.
+  > A _hash map_ allows you to associate a value with a particular key.
+
+- Emphasize important concepts or words
+
+  > When `s` comes _into_ scope, it is valid. It remains valid until it goes _out of_ scope.
+
```

Comment:
Some books, like [The Rust Programming Language](https://doc.rust-lang.org/book/ch02-00-guessing-game-tutorial.html#ensuring-reproducible-builds-with-the-cargolock-file), use italics for names of files and books.

Do you think the file names should be in monospace font as proposed in the Using `monospace` section?

Reply:
Then let's stick with the monospace font option, as the readability of italics is considered worse in general.